FEAT-#4371: Add logging to Modin (#4372)
modin-project/modin: docs/usage_guide/advanced_usage/modin_logging.rst

Logging and Tracing with Modin
==============================

> Modin logging and tracing allows users to gain observability into their queries by logging internal Modin API calls, partition metadata, 

Reviewer: Isn't this the exact same paragraph as above? Maybe change the above one to:
Modin also includes the ability to log internal API calls and metadata, to offer users greater insight into their queries. Please see our docs ... for more.

Author: I think it's fine for the descriptions to be the same here. The descriptions are the same for XGBoost and Spreadsheet API as well. It's a fairly short description, and I think that each line in the summary is important (what logging logs, default disabled, and that it is a local file).

and memory profiling. When Modin logging is enabled (default disabled), log files are written to a local `.modin` directory at the same
directory level as the notebook/script used to run Modin. Each log file in the `.modin` directory is named after the job uuid. 

Usage example
-------------

In the example below, we enable logging for internal Modin API calls. 

.. code-block:: python

  import modin.pandas as pd
  from modin.config import LogMode
  LogMode.enable_api_only()

  # Your code goes here

> In this example, we enable logging for internal Modin API calls, partition metadata, and memory profiling.

Reviewer: This gets a little repetitive since its essentially the same thing.
```suggestion
In the next example, we add logging not only for internal API calls, but also for partition metadata and memory profiling.```

We can set the granularity (in seconds) at which the system memory utilization is logged using `LogMemoryInterval`. 

.. code-block:: python

  import modin.pandas as pd
  from modin.config import LogMode, LogMemoryInterval
  LogMode.enable()

>   LogMemoryInterval.put(2) # Defaults to 5 seconds

Reviewer: ```suggestion
  LogMemoryInterval.put(2) # Default is 5 seconds - New Interval is 2 seconds
```
just to make it a little more clear what's happening here.

  # Your code goes here

> Modin logging is default disabled, but if currently enabled you can manually disable the logs as follows:

Reviewer: ```suggestion
Disable Modin logging like so:
```
